pkg/server/server.go

```diff
@@ -41,8 +49,38 @@ func Start(cfg Config) {
 		WriteTimeout: 40 * time.Second,
 	}
 
-	slog.Infof("listening on http://%s:%d", loopback, cfg.Port)
-	slog.Infof("targeting node %s", cfg.NodeIP)
+	network := "tcp"
+	addr := net.JoinHostPort(loopback, strconv.Itoa(cfg.Port))
+	if cfg.PreferUnixSocket {
+		network = "unix"
+		addr = cfg.UnixSocket
+	}
+
+	// gracefully shutdown HTTP server
+	idleConnClosed := make(chan struct{})
+	go func() {
+		sigCh := make(chan os.Signal, 1)
+		signal.Notify(sigCh, os.Interrupt, syscall.SIGTERM, syscall.SIGQUIT)
+		slog.Info("Received signal: ", <-sigCh)
+		if err := httpServer.Shutdown(context.Background()); err != nil {
```

Comment:
```suggestion
		sigCh := make(chan os.Signal, 1)
		signal.Notify(sigCh, os.Interrupt, syscall.SIGTERM, syscall.SIGQUIT)
		slog.Info("Received signal: ", <-sigCh)
		ctx, cancel := context.WithTimeout(context.Background(), 45*time.Second)
	        defer func() {
		        slog.Info("graceful shutdown timeout")
		        cancel()
	        }()
		if err := httpServer.Shutdown(ctx); err != nil {
```

Reply:
I was thinking something like this will give us a little bit more info than just waiting for k8s to SIGKILL the container. But upon checking, I don't see anything useful that can be logged here. (was thinking if we could log some info on the in-flight request we are waiting for)

Reply:
Well, probably digging into `net` or `net/http` packages can give some results (about active connections). But here we are entering quite a low level zone as the connections drain and active/inactive TCP connections is 1) thin ice we probably would not like to rely a lot on, 2) not an easy topic to dig.

Reply:
Yeah, I agree.